github-actions/scan/src/configuration-model-default-helper.ts

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+
+export const DEFAULT_EXCLUDES: string[] = ['**/.sechub-gha/**']
+
+/**
+ * Adds default exclude patterns to the provided JSON data, without altering the original JSON data.
+ * 
+ * If a 'data' section is present the default exclude patterns will be added to each entry's 'excludes' array (sources and binaries).
+ * In case the 'data' section is not present, the deprecated 'codeScan.excludes' will be checked instead.
+ * If no sources or binaries are specified at all, no default exclude patterns will be added.
+ * 
+ * @param sechubConfigJson - The JSON data to be updated.
+ * @returns A copy of the JSON data with the default exclude patterns added.
+ */
+export function addDefaultExcludesToSecHubConfig(sechubConfigJson: any): any {
+    let sechubConfigJsonCopy = { ...sechubConfigJson };
+
+    let dataSection = sechubConfigJsonCopy.data;
+    if (dataSection === undefined && sechubConfigJsonCopy.codeScan !== undefined) {
```

Comment:
Please reduce complexity/make more readable, maintainable:

E.g.
```typescript

handleDataSectionReferences()
handleLegacyCodeScanFileSystmeParts()

```
This makes it also clear that a mixup/combination of legacy + data section could be possible...

Reply:
having these methods, we can also do guard closes.

e.g.
```typescript
function handleDataSectionReferences(dataSection) {
   if (dataSection != undefined){
      return;
   }
   
   // do processing.
}
```